### PR TITLE
TSPS-348 part one: Publish Python thin client 

### PR DESCRIPTION
**Pull request by mmorgantaylor**
### Description 

Publish the thin (auto-generated) client to PyPi via GHA.
- PyPi project name: terra-scientific-pipelines-service-api-client
- PyPi package name: teaspoons_client

### Jira Ticket
achieves part of https://broadworkbench.atlassian.net/browse/TSPS-348

### CODE REVIEW

File: .github/workflows/release-python-client.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      - name: set version to 7.8.0
+        run: openapi-generator-cli version-manager set 7.8.0
+
+      - name: Generate Python client
+        run: |
+          openapi-generator-cli generate \
+          -i common/openapi.yml \
+          -g python \
+          -o teaspoons-client \
+          --additional-properties=projectName=terra-scientific-pipelines-service-client,packageName=teaspoons_client,packageVersion=${{ inputs.new-tag || '0.0.0' }},httpUserAgent=terra-scientific-pipeliens-service-client/${{ inputs.new-tag || '0.0.0'}}/python
```
Comment (jsotobroad) [inline .github/workflows/release-python-client.yml:41]: Do we call this `terra-scientific-pipelines-service-thin-client` or how will we distinguish this Pypi package from the thick client?

File: .github/workflows/tag-publish.yml [hunk 2]
```diff
@@ -22,15 +22,23 @@ on:
   push:
```
Comment (jsotobroad) [inline .github/workflows/tag-publish.yml:22]: ti think you may have forgotten to commit the latest changes to this file

File: .github/workflows/tag-publish.yml [hunk 3]
```diff
@@ -169,7 +182,7 @@ jobs:
         uses: ncipollo/release-action@v1
         id: create_release
         with:
-          tag: ${{ steps.tag.outputs.tag }}
```
Comment (jsotobroad) [inline .github/workflows/tag-publish.yml:172]: there is another place in this job step where `steps.tag.outputs.tag` is referenced, do we not need to change that as well? 

https://github.com/DataBiosphere/terra-scientific-pipelines-service/blob/main/.github/workflows/tag-publish.yml#L156

Comment (mmorgantaylor) [reply .github/workflows/tag-publish.yml:172]: oh good call, there were actually several other places that needed to be updated. thank you!

**Pull request merged**